react/components/ComponentEditor/index.tsx

```diff
@@ -334,26 +385,71 @@ class ComponentEditor extends Component<
         <Modal
           isActionLoading={this.state.isLoading}
           isOpen={this.state.isModalOpen}
-          onClickAction={this.handleConfigurationSave}
-          onClickCancel={this.handleConfigurationDiscard}
-          onClose={this.handleModalClose}
-          textButtonAction={intl.formatMessage({
-            id: 'pages.editor.components.button.save',
-          })}
-          textButtonCancel={intl.formatMessage({
-            id: 'pages.editor.components.modal.button.discard',
-          })}
-          textMessage={intl.formatMessage({
-            id: 'pages.editor.components.modal.text',
-          })}
+          onClickAction={
+            this.state.isDeleteModalOpen
+            ?
+            this.handleConfigurationDelete
+            :
+            this.handleConfigurationSave
+          }
+          onClickCancel={
+            this.state.isDeleteModalOpen
+            ?
+            this.handleModalClose
+            :
+            this.handleConfigurationDiscard
+          }
+          onClose={
+            this.state.isDeleteModalOpen
+            ?
+            this.handleModalClose
+            :
+            this.handleModalClose
```

Comment:
This ternary is assigning the same function in both cases

Reply:
ok